### PR TITLE
 chore(config): [Multisafepay] Add configs for card mandates for Multisafepay

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## How did you test it?

Test credentials not available, therefore can't be tested

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 1]
```diff
@@ -390,7 +395,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for MultisafepayPaymentsReques
             recurring_model: if item.request.setup_future_usage
                 == Some(enums::FutureUsage::OffSession)
             {
-                Some("Unscheduled".to_string())
+                Some("unscheduled".to_string())
```
Comment (ArjunKarthik) [inline crates/router/src/connector/multisafepay/transformers.rs:398]: @AkshayaFoiger Let's have a enum for recurring_model

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 2]
```diff
@@ -390,7 +395,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for MultisafepayPaymentsReques
             recurring_model: if item.request.setup_future_usage
                 == Some(enums::FutureUsage::OffSession)
```
Comment (ArjunKarthik) [inline crates/router/src/connector/multisafepay/transformers.rs:396]: recurring_model should be based on setup_mandate_details and not item.request.setup_future_usage

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 3]
```diff
@@ -305,7 +309,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for MultisafepayPaymentsReques
             email: item.request.email.clone(),
             user_agent: None,
             referrer: None,
-            reference: None,
+            reference: Some(item.attempt_id.clone()),
```
Comment (ArjunKarthik) [inline crates/router/src/connector/multisafepay/transformers.rs:312]: Let's do some cleanup, Keep only necessary field in the struct

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — DISMISSED** [review]
@AkshayaFoiger Let's keep only the minimum fields in the request and response struct. Please take this in the next PR

File: crates/router/src/connector/multisafepay/transformers.rs (hunk 3 above)
Comment (prasunna09) [inline crates/router/src/connector/multisafepay/transformers.rs:312]:
```suggestion
            reference: Some(item.router_data.connector_request_reference_id.clone()),
```

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]

**Pull request merged**